src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -313,9 +352,27 @@ private float addRow(
           RowBufferStats.getCombinedStats(statsMap.get(columnName), forkedColStats.getValue()));
     }
 
-    // Increment null count for column missing in the input map
+    // Increment null count for column and its sub-columns missing in the input map
     for (String columnName : Sets.difference(this.fieldIndex.keySet(), inputColumnNames)) {
-      statsMap.get(columnName).incCurrentNullCount();
+      if (clientBufferParameters.getIsIcebergMode()) {
+        if (subColumnFinder == null) {
+          throw new SFException(ErrorCode.INTERNAL_ERROR, "SubColumnFinder is not initialized.");
+        }
+        subColumnFinder
+            .getSubColumns(columnName)
+            .forEach(
+                subColumn -> {
```

Comment:
nit: you can save four levels of indentation by doing `for (String subColumn: subColumnFilder.getSubColumns(columnName)) {`